contracts/ERC20MultiDelegate.sol

```diff
@@ -104,19 +115,19 @@ contract ERC20MultiDelegate is ERC1155, Ownable {
             "Delegate: The number of amounts must be equal to the greater of the number of sources or targets"
         );
 
+        uint256 maxLength = Math.max(sourcesLength, targetsLength);
         // Iterate until all source and target delegates have been processed.
-        for (
-            uint transferIndex = 0;
-            transferIndex < Math.max(sourcesLength, targetsLength);
-        ) {
+        for (uint transferIndex = 0; transferIndex < maxLength; ) {
             address source = address(0);
             address target = address(0);
             if (transferIndex < sourcesLength) {
-                require((sources[transferIndex] >> 160) == 0, "Upper 96 bits of source uint256 must be zero");
+                if ((sources[transferIndex] >> 160) != 0)
+                    revert Upper96BitsNotZero();
```

Comment:
This should probably be less implementation-specific, like `InvalidDelegateAddress`.